### PR TITLE
preserve camel case on additional ingress labels

### PR DESCRIPTION
**Pull request by gurfm**
Ensure camel case is preserved on additional ingress labels provided in the Kiali CR.

Fixes https://github.com/kiali/kiali/issues/7145

### CODE REVIEW

**Comment by jmazzitelli** [comment]
@gurfm can you write a github issue so this PR is tracked with an issue. Please refer to the community contribution guidelines that talk about this requirement: https://github.com/kiali/kiali/blob/master/CONTRIBUTING.md#making-a-change

**Comment by jmazzitelli** [comment]
FWIW: this code looks good. Thanks for finding this. Let's get a issue written up and we can finish testing and getting this merged.

**Comment by gurfm** [comment]
> FWIW: this code looks good. Thanks for finding this. Let's get a issue written up and we can finish testing and getting this merged.

Sure here it is: https://github.com/kiali/kiali/issues/7145

Could you please consider backporting this to the Kiali version that will be used with Maistra 2.5 (I don't know which one it will be)?

**Comment by jmazzitelli** [comment]
> Could you please consider backporting this to the Kiali version that will be used with Maistra 2.5 (I don't know which one it will be)?

It would be v1.73. We are late in the cycle, might have to push this to OSSM 2.5.1. I'm running into a test error when I run the config-values-test molecule test that this PR modifies. But the error seems unrelated. I'm trying to figure out what the issue is now.

File: roles/default/kiali-deploy/tasks/snake_camel_case.yaml
Comment (gurfm) [reply roles/default/kiali-deploy/tasks/snake_camel_case.yaml:61]: in fact I had trouble with the local setup to run tests so I was not able to validate locally. I validated however the patch for the 1.65 version (which is slightly different) directly on our Openshift cluster:

```
- name: Replace snake_case with camelCase in deployment.ingress.additional_labels
  set_fact:
    kiali_vars: |
      {% set a=kiali_vars['deployment']['ingress'].pop('additional_labels') %}
      {{ kiali_vars | combine({'deployment': {'ingress': {'additional_labels': current_cr.spec.deployment.ingress.additional_labels }}}, recursive=True) }}
  when:
  - kiali_vars.deployment.ingress.additional_labels is defined
  - kiali_vars.deployment.ingress.additional_labels | length > 0
```

Comment (jmazzitelli) [reply roles/default/kiali-deploy/tasks/snake_camel_case.yaml:61]: OK, I found it ... that wasn't easy to find. blank space at the end of line here breaks things. I'll make that change in my PR I submit.
```suggestion
      {# deployment.ingress.additional_labels #}
      {% if kiali_vars.deployment.ingress.additional_labels is defined and kiali_vars.deployment.ingress.additional_labels | length > 0 %}
      {%   set _=kiali_vars['deployment']['ingress'].pop('additional_labels') %}
      {%   set kiali_vars=kiali_vars | combine({'deployment': {'ingress': {'additional_labels': current_cr.spec.deployment.ingress.additional_labels }}}, recursive=True) %}
      {% endif %}
      {# #}
```

Comment (gurfm) [reply roles/default/kiali-deploy/tasks/snake_camel_case.yaml:61]: > OK, I found it ... that wasn't easy to find. blank space at the end of line here breaks things. I'll make that change in my PR I submit.

thank you for spotting this :)

**Comment by jmazzitelli** [comment]
@gurfm you'll need to merge my PR into yours: https://github.com/gurfm/kiali-operator/pull/1

I was able to test successfully on both minikube and openshift. On OpenShift, this molecule test passes:

```
make -e MOLECULE_SCENARIO="affinity-tolerations-resources-test" -e MOLECULE_USE_DEV_IMAGES="true" build build-ui cluster-push molecule-test
```

**Comment by jmazzitelli** [comment]
I put the DNM label on this PR - do not merge until my PR has been merged into this one.

**Comment by jmazzitelli** [comment]
the updated molecule test passes on both minikube and openshift

**Review by jmazzitelli — APPROVED** [review]

**Pull request merged**
